xbmc/dialogs/GUIDialogBusy.h

```diff
@@ -46,9 +36,17 @@ class CGUIDialogBusy: public CGUIDialog
    \return true if the event completed, false if cancelled.
    */
   static bool WaitOnEvent(CEvent &event, unsigned int displaytime = 100, bool allowCancel = true);
-protected:
+
+private:
+  CGUIDialogBusy();
+  ~CGUIDialogBusy() override;
+
   void Open_Internal(bool bProcessRenderLoop, const std::string& param = "") override;
-  bool m_bCanceled;
-  bool m_bLastVisible = false;
-  float m_progress; ///< current progress
+  bool OnBack(int actionID) override;
+  void DoProcess(unsigned int currentTime, CDirtyRegionList& dirtyregions) override;
+  void Render() override;
+
+  bool m_bLastVisible{false};
+  bool m_cancelled{false};
+  uint32_t m_waiters{0};
```

Comment:
shouldn't this be an atomic integer instead?

Reply:
I don't think so. `CGUIDialogBusy::WaitOnEvent` must be called exclusively from the rendering (the main) thread.

Reply:
Yeah you're right. But this is not that obvious any way it can be made safer? Maybe throwing a logic error if `waitOnEvent` is called out of the rendering thread (ref https://github.com/xbmc/xbmc/pull/21441)?

Reply:
I actually used exactly this safety net (exception) while doing runtime tests for this change, but removed it because I needed to include Application.h for this to check the current thread. If this new dependency would be okay I could re-introduce the check/exception.

Reply:
> CGUIDialogBusy::WaitOnEvent must be called exclusively from the rendering
Where does this info come from?

https://github.com/xbmc/xbmc/blob/9f699af44bde1b8b3448c3d4162fbb089974d38e/xbmc/pvr/windows/GUIWindowPVRGuide.cpp#L887